xmtp_mls/IDENTITY.md

```diff
@@ -0,0 +1,82 @@
+# XMTP Identity Structure
+
+In XMTP v3, a messaging account is represented by an Ethereum wallet address. An account consists of multiple app installations that may send and receive messages on behalf of it. Each installation is a separate cryptographic identity with its own set of keys.
+
+```
+Amal's account (Ethereum wallet address)
+│
+├── Converse app (mobile phone)
+│   └── Installation key bundle 1
+│
+├── Coinbase Wallet (mobile phone)
+│   └── Installation key bundle 2
+│
+├── Lenster (tablet)
+│   └── Installation key bundle 3
+│
+└── Coinbase Wallet (tablet)
+    └── Installation key bundle 4
+```
+
+Using per-installation keys provides the following benefits:
+
+- Installation private keys are never shared across devices or published onto the network.
+- The user may enumerate the installations that have messaging access to their account.
+- The user may revoke keys on a per-installation level.
+
+**Installation provisioning**
+
+Every new app installation gains messaging access as follows:
+
+1. A new Ed25519 signature key pair is generated and stored on the device, representing the installation's identity.
+2. The app prompts the user to sign the public key with their Ethereum wallet, establishing an association between the installation's identity and the user’s account. Example text:
+
+   ```
+   XMTP: Grant Messaging Access
+
+   Current Time: <current time and local timezone>
+   Installation Key: <hex(last_20_bytes(keccak256(Ed25519PublicKey)))>
+   ```
+
+3. The following data is protobuf-serialized to form the MLS Credential. The credential can be presented alongside the installation key to prove an association with an account:
+
+   ```
+   MlsCredential {
+       Eip191Association {
+           association_text_version: i32,
+           signature: bytes,
+           wallet_address: string,
+           creation_time_ns: i64,
+       }
+   }
+   ```
+
+4. A last resort KeyPackage (signed by the installation key per the MLS spec) is generated and stored on the device.
+5. The app publishes the public signing key, credential, and last resort key package to the account's contact topic on the server. This enables other apps to understand that the installation is on the network, associated with the account, and how to contact it.
+
+**Installation management**
+
+At any time, the user may enumerate active installations by querying the contact topic for the account. The user may identify each installation by the creation time as well as the installation key from the signing text.
```

Comment:
We no longer have a contact topic, and in MLS all installations are stored in the MLS db but not on the network